catalog/security/catalog-security-logging/src/main/java/org/codice/ddf/catalog/security/logging/SecurityLoggingPlugin.java

```diff
@@ -13,85 +13,260 @@
+        return input;
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        String additional = input.getCreatedMetacards()
+                .stream()
+                .map(Metacard::getId)
+                .collect(Collectors.joining(",", "Created metacards: ", ""));
+        logOperation(CatalogOperationType.INGEST_RESPONSE, input, additional);
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        String additional = input.getUpdatedMetacards()
+                .stream()
+                .map(Update::getNewMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(",", "Updated metacards: ", ""));
+        logOperation(CatalogOperationType.UPDATE_RESPONSE, input, additional);
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        String additional = input.getDeletedMetacards()
+                .stream()
+                .map(Metacard::getId)
+                .collect(Collectors.joining(",", "Deleted metacards: ", ""));
+        logOperation(CatalogOperationType.DELETE_RESPONSE, input, additional);
+        return input;
+    }
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        String addtional = input.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(",", "Queried metacards: ", ""));
+        logOperation(CatalogOperationType.QUERY_RESPONSE, input, addtional);
+        return input;
+    }
+
+    @Override
+    public ResourceResponse process(ResourceResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        String addtional = "Return resource: " + input.getResource()
+                .getName();
+        logOperation(CatalogOperationType.RESOURCE_RESPONSE, input, addtional);
         return input;
     }
 
-    private void logOperation(CatalogOperationType operationType, Request request) {
-        String user;
-        try {
-            Subject subject = SecurityUtils.getSubject();
-            user = SubjectUtils.getName(subject, NO_USER);
-        } catch (Exception e) {
-            user = NO_USER;
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        String additional = "";
+        Query query = input.getQuery();
+        if (query instanceof QueryImpl) {
+            additional = ((QueryImpl) query).getFilter()
+                    .toString();
         }
+        logOperation(CatalogOperationType.QUERY_REQUEST, input, additional);
```

Comment:
Do we want to include the source information in this as well? and those following?